app/controllers/api/traces_controller.rb

```diff
@@ -12,7 +13,14 @@ class TracesController < ApiController
     def show
       @trace = Trace.visible.find(params[:id])
 
-      head :forbidden unless @trace.public? || @trace.user == current_user
+      if @trace.public? || @trace.user == current_user
+        respond_to do |format|
+          format.xml
+          format.json
+        end
+      else
+        head :forbidden
+      end
```

Comment:
```suggestion
      return head :forbidden unless @trace.public? || @trace.user == current_user

      respond_to do |format|
        format.xml
        format.json
      end
```
I'd keep the `head :forbidden unless` guard clause from before